src/Link.tsx

```diff
@@ -2,21 +2,31 @@ import useEventCallback from '@restart/hooks/useEventCallback';
 import React from 'react';
 import warning from 'tiny-warning';
 
+import { LinkInjectedProps, LinkProps } from './typeUtils';
 import useRouter from './useRouter';
 
+// TODO: move this to utils
+// Workaround for TS bug where function is not callable https://github.com/microsoft/TypeScript/issues/37663
```

Comment:
If we simply checked `typeof someVariable === 'function'` we'd get this :/ https://twitter.com/SzymonWiszczuk/status/1607820009736396800

Reply:
Actually, since I decided to `any` props in Link, this is not needed anymore(cause `children` is `any`) 

This will definitely come up again once I get back to this component(I plan to do that after I finish porting all the files)